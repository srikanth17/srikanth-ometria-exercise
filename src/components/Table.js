import React from 'react';

const Table = ({ characters }) => {
    return (
        <table className="highlight">
            <thead>
            <tr>
                <th>Name</th>
                <th>Height</th>
                <th>Gender</th>
            </tr>
            </thead>

            <tbody>
            {characters && characters.map((character) => (
                <tr key={character.name}>
                    <td>{character.name}</td>
                    <td>{character.height}</td>
                    <td>{character.gender}</td>
                </tr>
            ))}
            </tbody>
        </table>
    );
};

export default Table;