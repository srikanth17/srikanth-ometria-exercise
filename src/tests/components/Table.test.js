import React from 'react'
import { shallow } from 'enzyme'
import Table from '../../components/Table'

test('should render Table correctly', () => {
    const wrapper = shallow(<Table/>);
    expect(wrapper).toMatchSnapshot();
});