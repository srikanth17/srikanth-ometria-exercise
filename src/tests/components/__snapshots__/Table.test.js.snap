// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Table correctly 1`] = `
<table
  className="highlight"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Height
      </th>
      <th>
        Gender
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
